windows/RNFS/RNFSManager.cpp

```diff
@@ -0,0 +1,1115 @@
+        std::filesystem::path path(hasTrailingSlash ? filepath.substr(0, pathLength - 1) : filepath);
+        path.make_preferred();
+
+        StorageFolder folder{ co_await StorageFolder::GetFolderFromPathAsync(path.parent_path().wstring()) };
+        auto target{ co_await folder.GetItemAsync(path.filename().wstring()) };
+        co_await target.DeleteAsync();
+
+        promise.Resolve();
+    }
+}
+catch (const hresult_error& ex)
+{
+    hresult result{ ex.code() };
+    if (result == 0x80070002) // FileNotFoundException
+    {
+        promise.Reject(RN::ReactError{ "ENOENT", "ENOENT: no such file or directory, open " + filepath });
+    }
+    else
+    {
+        // "Failed to unlink file" 
+        promise.Reject( winrt::to_string(ex.message()).c_str() );
+    }
+}
+
+
+winrt::fire_and_forget RNFSManager::exists(std::string filepath, RN::ReactPromise<bool> promise) noexcept
+try
+{
+    size_t fileLength{ filepath.length() };
+
+    if (fileLength <= 0) {
+        promise.Resolve(false);
+    }
+    else {
+        bool hasTrailingSlash{ filepath[fileLength - 1] == '\\' || filepath[fileLength - 1] == '/' };
+        std::filesystem::path path(hasTrailingSlash ? filepath.substr(0, fileLength - 1) : filepath);
+
+        winrt::hstring directoryPath, fileName;
+        splitPath(filepath, directoryPath, fileName);
+        StorageFolder folder{ co_await StorageFolder::GetFolderFromPathAsync(directoryPath) };
+        if (fileName.size() > 0) {
+            co_await folder.GetItemAsync(fileName);
+        }
+        promise.Resolve(true);
+    }
+}
+catch (const hresult_error& ex)
+{
+    hresult result{ ex.code() };
+    if (result == 0x80070002) {
+        promise.Resolve(false);
+    }
+    // "Failed to check if file or directory exists.
+    promise.Reject(winrt::to_string(ex.message()).c_str());
+}
+
+
+void RNFSManager::stopDownload(int32_t jobID) noexcept
+{
+    m_tasks.Cancel(jobID);
+}
+
+
+void RNFSManager::stopUpload(int32_t jobID) noexcept
+{
+    m_tasks.Cancel(jobID);
+}
+
+
+winrt::fire_and_forget RNFSManager::readFile(std::string filepath, RN::ReactPromise<std::string> promise) noexcept
+try
+{
+    winrt::hstring directoryPath, fileName;
+    splitPath(filepath, directoryPath, fileName);
+
+    StorageFolder folder{ co_await StorageFolder::GetFolderFromPathAsync(directoryPath) };
+    StorageFile file{ co_await folder.GetFileAsync(fileName) };
+
+    Streams::IBuffer buffer{ co_await FileIO::ReadBufferAsync(file) };
+    winrt::hstring base64Content{ Cryptography::CryptographicBuffer::EncodeToBase64String(buffer) };
+    promise.Resolve(winrt::to_string(base64Content));
+}
+catch (const hresult_error& ex)
+{
+    hresult result{ ex.code() };
+    if (result == 0x80070002) // FileNotFoundException
+    {
+        promise.Reject(RN::ReactError{ "ENOENT", "ENOENT: no such file or directory, open " + filepath });
+    }
+    else if (result == 0x80070005) // UnauthorizedAccessException
```

Comment:
`E_ACCESSDENIED`

Reply:
Just pushed a commit that implements that change.